generators/typescript/sdk/client-class-generator/src/auth-provider/RoutingAuthProviderGenerator.ts

```diff
@@ -0,0 +1,259 @@
+                                    ts.factory.createPropertySignature(
+                                        undefined,
+                                        "endpointMetadata",
+                                        ts.factory.createToken(ts.SyntaxKind.QuestionToken),
+                                        context.coreUtilities.fetcher.EndpointMetadata._getReferenceToType()
+                                    )
+                                ])
+                            )
+                        }
+                    ],
+                    returnType: getTextOfTsNode(
+                        ts.factory.createTypeReferenceNode("Promise", [
+                            context.coreUtilities.auth.AuthRequest._getReferenceToType()
+                        ])
+                    ),
+                    statements: this.generateGetAuthRequestStatements()
+                }
+            ]
+        });
+    }
+
+    private generateGetAuthRequestStatements(): string {
+        const providerMap = `this.${AUTH_PROVIDERS_FIELD_NAME}`;
+
+        return `
+        const security = arg?.endpointMetadata?.security;
+
+        // If no security requirements specified, endpoint is anonymous - return empty auth request
+        if (security == null || security.length === 0) {
+            return { headers: {} };
+        }
+
+        // First, verify that at least one security requirement can be satisfied by available providers
+        const canSatisfyAnyRequirement = security.some((securityRequirement) => {
+            const schemeKeys = Object.keys(securityRequirement);
+            return schemeKeys.every((schemeKey) => ${providerMap}.has(schemeKey));
+        });
+
+        if (!canSatisfyAnyRequirement) {
+            const requiredSchemes = security.map((req) => Object.keys(req).join(" AND ")).join(" OR ");
+            throw new Error(
+                \`No authentication credentials provided that satisfy the endpoint's security requirements. ` +
+                \`Required: \${requiredSchemes}. Please provide the necessary authentication credentials.\`
+            );
+        }
+
+        // Try each security requirement collection (OR relationship)
+        for (const securityRequirement of security) {
+            const schemeKeys = Object.keys(securityRequirement);
+            let allSchemesSucceeded = true;
+            const combinedHeaders: Record<string, string> = {};
+
+            // Try all schemes in this requirement (AND relationship)
+            for (const schemeKey of schemeKeys) {
+                const provider = ${providerMap}.get(schemeKey);
+                if (provider == null) {
+                    allSchemesSucceeded = false;
+                    break;
+                }
+
+                try {
+                    const authRequest = await provider.getAuthRequest(arg);
+                    if (Object.keys(authRequest.headers).length === 0) {
```

Comment:
When the resulting auth request has 0 headers, that does not equate to failure. If no error is thrown, it was successful.